lib/Mojolicious/Validator/Validation.pm

```diff
@@ -23,10 +23,12 @@ sub check {
 
   return $self unless $self->is_valid;
 
-  my $cb    = $self->validator->checks->{$check};
-  my $name  = $self->topic;
-  my $input = $self->input->{$name};
+  my $cb          = $self->validator->checks->{$check};
+  my $name        = $self->topic;
+  my $input       = $self->input->{$name};
+  my $is_optional = $self->is_optional($name);
```

Comment:
```$is_optional``` will always be true here. Not sure if you expect to fetch the old value or what's going on here :disappointed: